services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,33 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.{
+  ProjectConfigurationRequest,
+  ProjectConfigurationResponse
+}
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationRequest): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .updateOne(
+        toDocument(ProjectConfiguration(configuration.projectId, configuration.projectFileConfigurations)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationResponse]] =
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).flatMap {
```

Comment:
Do we need `flatMap` here? I think `map` would be more appropriate.

Reply:
Like this?
``` repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).map(optionConf => optionConf.map(conf => if (conf.isActive) ProjectConfigurationResponse(conf.projectId, conf.projectFileConfigurations) else throw Exception))```